src/tests/test_crystals.cpp

```diff
@@ -0,0 +1,483 @@
+                            random_encoding_roundtrips<Kyberish_Trait, 42>(res, *rng, 6);
+                            random_encoding_roundtrips<Kyberish_Trait, 128>(res, *rng, 8);
+                            random_encoding_roundtrips<Kyberish_Trait, 1337>(res, *rng, 11);
+                         }),
+
+      Botan_Tests::CHECK("random encoding roundtrips (Kyber ranges)",
+                         [](Test::Result& res) {
+                            auto rng = Test::new_rng("CRYSTALS encoding roundtrips as used in kyber");
+                            random_encoding_roundtrips<Kyberish_Trait, 1>(res, *rng, 1);
+                            random_encoding_roundtrips<Kyberish_Trait, (1 << 4) - 1>(res, *rng, 4);
+                            random_encoding_roundtrips<Kyberish_Trait, (1 << 5) - 1>(res, *rng, 5);
+                            random_encoding_roundtrips<Kyberish_Trait, (1 << 10) - 1>(res, *rng, 10);
+                            random_encoding_roundtrips<Kyberish_Trait, (1 << 11) - 1>(res, *rng, 11);
+                            random_encoding_roundtrips<Kyberish_Trait, Kyberish_Constants::Q - 1>(res, *rng, 12);
+                         }),
+
+      Botan_Tests::CHECK("random encoding roundtrips (Dilithium ranges)",
+                         [](Test::Result& res) {
+                            using Dilithiumish_Trait = Mock_Trait<Dilithiumish_Constants>;
+
+                            auto rng = Test::new_rng("CRYSTALS encoding roundtrips as used in kyber");
+                            constexpr auto t1 = 1023;
+                            constexpr auto gamma2_32 = 15;
+                            constexpr auto gamma2_88 = 43;
+                            constexpr auto gamma1_17 = 131072;
+                            constexpr auto gamma1_19 = 524288;
+                            constexpr auto eta2 = 2;
+                            constexpr auto eta4 = 4;
+                            constexpr auto twotothed = 4096;
+                            random_encoding_roundtrips<Dilithiumish_Trait, t1>(res, *rng, 10);
+                            random_encoding_roundtrips<Dilithiumish_Trait, gamma2_32>(res, *rng, 4);
+                            random_encoding_roundtrips<Dilithiumish_Trait, gamma2_88>(res, *rng, 6);
+                            random_encoding_roundtrips<Dilithiumish_Trait, 2 * gamma1_17 - 1>(res, *rng, 18);
+                            random_encoding_roundtrips<Dilithiumish_Trait, 2 * gamma1_19 - 1>(res, *rng, 20);
+                            random_encoding_roundtrips<Dilithiumish_Trait, 2 * eta2>(res, *rng, 3);
+                            random_encoding_roundtrips<Dilithiumish_Trait, 2 * eta4>(res, *rng, 4);
+                            random_encoding_roundtrips<Dilithiumish_Trait, 2 * twotothed - 1>(res, *rng, 13);
+                         }),
+   };
+}
+
+}  // namespace
+
+BOTAN_REGISTER_TEST_FN("pubkey", "CRYSTALS", test_extended_euclidean_algorithm, test_polynomial_basics, test_encoding);
```

Comment:
```suggestion
BOTAN_REGISTER_TEST_FN("pubkey", "crystals", test_extended_euclidean_algorithm, test_polynomial_basics, test_encoding);
```
it seems all other test suits use snake_case (see `./botan-test --help`). Maybe, while you're at it, adapt the names for dilithium as well (e.g.`dilithium_kat_4x4_AES_Deterministic`, `dilithium_kat_4x4_Randomized`, etc.).

Reply:
Fair. Regarding the dilithium KATs: These names are currently tightly coupled with  the `DilithiumMode` constants. 😨 This will need a cleanup anyway, but lets not do that in this PR.